src/Symfony/Bundle/FrameworkBundle/Command/ServerStartCommand.php

```diff
@@ -177,25 +182,48 @@ private function isOtherServerProcessRunning($address)
     }
 
     /**
+     * Determine the absolute file path for the router script based on user input and the environment.
+     *
+     * @param InputInterface  $input  An InputInterface instance
+     * @param OutputInterface $output An OutputInterface instance
+     * @param string          $env    The application environment
+     *
+     * @return string|bool The absolute file path of the router script, or false on failure
+     */
+    private function determineRouterScript(InputInterface $input, OutputInterface $output, $env)
+    {
+        $router = $input->getOption('router');
+
+        if (null === $router) {
+            $router = $this
+                ->getContainer()
+                ->get('kernel')
+                ->locateResource(sprintf('@FrameworkBundle/Resources/config/router_%s.php', $env))
+            ;
+        } else {
+            if (!file_exists($router)) {
+                $output->writeln(sprintf('<error>The given router script "%s" does not exist</error>', $router));
```

Comment:
I did consider logging an error if this function returned false, but I think it will always be superfluous, because the only two 'bad file path scenarios' I'm aware of are:
1. The `--router` argument was set to a file path which is found to not exist by the `file_exists` check on line 204, at this point the error is logged
2. No custom router script was specified, but the chosen environment does not have a corresponding standard router script (i.e. _@FrameworkBundle/Resources/config/router_prod.php_ or _@FrameworkBundle/Resources/config/router_dev.php_), in which case the `locateResource` function throws an `InvalidArgumentException`

So in both the above scenarios, the error should be clear to the user.

Potentially I could add a check to verify that `realpath` didn't return `false` before returning from the function, but personally think that's a little over-defensive.  It would also mean assigning the result to a separate variable, so that the original can be logged.